### PR TITLE
Use connection ID as key for WebTransportSessions.

### PR DESCRIPTION
**Pull request by jianjunz**
WebTrannsportSessionId is not unique for all the sessions associated
with a certain server. It starts from 0 for each connection. The old
session is destoryed when a new session with the same ID is created.

As we only allow a single WebTransport session for a QUIC connection,
it's safe to use QUIC connection ID as key.

### CODE REVIEW

File: web_transport/sdk/impl/web_transport_server_backend.h [hunk 1]
```diff
@@ -40,8 +40,8 @@ class WebTransportServerBackend : public WebTransportSessionVisitor {
 
  private:
   WebTransportServerInterface::Visitor* visitor_;
-  std::unordered_map<::quic::WebTransportSessionId,
-                     std::unique_ptr<WebTransportServerSession>>
+  // Key is HTTP/3 QUIC stream connection ID.
+  std::unordered_map<std::string, std::unique_ptr<WebTransportServerSession>>
```
Comment (taste1981) [inline web_transport/sdk/impl/web_transport_server_backend.h:44]: make sure you guard access to this map.

File: web_transport/sdk/impl/web_transport_server_backend.cc [hunk 2]
```diff
@@ -32,9 +32,15 @@ void WebTransportServerBackend::OnSessionReady(
...
       std::make_unique<WebTransportServerSession>(session, http3_session,
                                                   io_runner_, event_runner_);
-  sessions_[session->id()] = std::move(wt_session);
+  WebTransportServerSession* session_ptr = wt_session.get();
+  if (sessions_.count(http3_session->connection_id().ToString()) > 0) {
+    LOG(WARNING) << "Session with the same connection ID exits, the old one "
+                    "will be terminated. Only one WebTransport session for a "
+                    "QUIC connection is supported.";
+  }
+  sessions_[http3_session->connection_id().ToString()] = std::move(wt_session);
```
Comment (taste1981) [inline web_transport/sdk/impl/web_transport_server_backend.cc:41]: Where do you remove the session id from the map?

File: web_transport/sdk/impl/web_transport_server_backend.h (hunk 1 above)
Comment (jianjunz) [reply web_transport/sdk/impl/web_transport_server_backend.h:44]: It's only called when headers are received, which is expected to be called on the same thread. So no lock is checked here.

File: web_transport/sdk/impl/web_transport_server_backend.cc (hunk 2 above)
Comment (jianjunz) [reply web_transport/sdk/impl/web_transport_server_backend.cc:41]: It's not being removed. I'll file an issue and fix it later.

File: web_transport/sdk/impl/web_transport_server_backend.h (hunk 1 above)
Comment (taste1981) [reply web_transport/sdk/impl/web_transport_server_backend.h:44]: you will need this eventually as you'll remove ids from the map.

Comment (jianjunz) [reply web_transport/sdk/impl/web_transport_server_backend.h:45]: For what I can imagine at this time, an ID is removed when a session is terminated by either sending a FIN, or receiving a FIN. Both of them are expected to be called on IO thread, which is the same thread for receiving headers. A thread checker is added to ensure session map is accessed on the same thread.

**Review by taste1981 — APPROVED** [review]

**Pull request merged**